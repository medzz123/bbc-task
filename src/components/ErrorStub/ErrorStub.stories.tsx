import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import ErrorStub from './ErrorStub';

export default {
  title: 'Components/ErrorStub',
  component: ErrorStub,
} as Meta;

const Template: Story = (storyArguments) => <ErrorStub {...storyArguments} />;

export const Default = Template.bind({});
