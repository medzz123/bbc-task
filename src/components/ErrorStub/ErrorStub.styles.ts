import { styled } from '@theme/config';

export const ErrorStubContainer = styled('button', {
  backgroundColor: '$main',
  color: '$background',
  padding: '$0',
  size: '$32',
  br: '$sizes$half',
  center: 'row',
  position: 'fixed',
  top: '$24',
  right: '$24',
  cursor: 'pointer',
  outline: 'none',
  border: 'none',
  transition: 'all 200ms ease',
  '&:hover, &:focus': {
    boxShadow: '0 0 0 $sizes$2 $colors$error',
  },

  variants: {
    isError: {
      error: {
        backgroundColor: '$error',
        '&:hover, &:focus': {
          boxShadow: '0 0 0 $sizes$2 $colors$success',
        },
      },
      success: {
        backgroundColor: '$success',
        '&:hover, &:focus': {
          boxShadow: '0 0 0 $sizes$2 $colors$error',
        },
      },
    },
  },
});
