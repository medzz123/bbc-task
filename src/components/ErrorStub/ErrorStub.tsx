import Icon from '@components/Icon';
import React, { FunctionComponent, useEffect, useState } from 'react';

import { ErrorStubContainer } from './ErrorStub.styles';

const ErrorStub: FunctionComponent = () => {
  const [error, setError] = useState(false);

  useEffect(() => {
    const item = localStorage.getItem('x-fail');

    setError(item === 'true' ? true : false);
  }, []);

  return (
    <ErrorStubContainer
      type="button"
      isError={error ? 'error' : 'success'}
      data-testid="errorStub"
      onClick={() => {
        const item = localStorage.getItem('x-fail');

        if (item === 'true') {
          setError(false);
          localStorage.setItem('x-fail', 'false');
        } else {
          setError(true);
          localStorage.setItem('x-fail', 'true');
        }
      }}
    >
      <Icon name={error ? 'danger' : 'smile'} />
    </ErrorStubContainer>
  );
};

export default ErrorStub;
