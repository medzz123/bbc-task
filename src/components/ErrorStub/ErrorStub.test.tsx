import { render } from '@test/library';
import React from 'react';

import ErrorStub from './ErrorStub';

test('ErrorStub Test', () => {
  const { container } = render(<ErrorStub />);

  expect(container).toMatchSnapshot();
});
