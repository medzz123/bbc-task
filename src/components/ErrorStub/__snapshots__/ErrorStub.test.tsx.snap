// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorStub Test 1`] = `
<div>
  <button
    class="sxsfhg3 sxsfhg3sbyv7--isError-success"
    data-testid="errorStub"
    type="button"
  >
    <i
      class="sx03kze sx03kze-uzmqd gg-smile"
    />
  </button>
</div>
`;
